src/modules/Search/Search.js

```diff
@@ -0,0 +1,698 @@
+      resultsClasses: 'visible',
+    })
+  }
+
+  close = () => {
+    debug('close()')
+    debug(`dropdown is ${!this.state.open ? 'already' : 'not yet'} closed`)
+    if (!this.state.open) return
+
+    this.trySetState({
+      open: false,
+    }, {
+      searchClasses: '',
+      resultsClasses: '',
+    })
+  }
+
+  // ----------------------------------------
+  // Render
+  // ----------------------------------------
+
+  renderSearchInput = () => {
+    const { icon, placeholder } = this.props
+    const { value } = this.state
+
+    return (
+      <Input
+        value={value}
+        placeholder={placeholder}
+        onBlur={this.handleBlur}
+        onChange={this.handleSearchChange}
+        onFocus={this.handleFocus}
+        // TODO: Input should probably add this class automatically when an icon
+        // is present. Gonna hold off until the V1 upgrade of the Input component.
+        className={icon ? 'icon' : ''}
+        inputClassName={'prompt'}
+        autoComplete='off'
+        tabIndex='0'
+        icon={icon}
+      />
+    )
+  }
+
+  renderNoResults = () => {
+    const { noResultsMessage, noResultsDescription, showNoResults } = this.props
+
+    if (!showNoResults) {
+      this.close()
+      return null
+    }
+
+    return (
+      <div className='message empty'>
+        <div className='header'>{noResultsMessage}</div>
+        {noResultsDescription &&
+          <div className='description'>{noResultsDescription}</div>
+        }
+      </div>
+    )
+  }
+
+  /**
+   * Offset is needed for determining the active item for results within a
+   * category. Since the index is reset to 0 for each new category, an offset
+   * must be passed in.
+   */
+  renderResult = (result, index, _array, offset = 0) => {
+    const { resultRenderer } = this.props
+    const { selectedIndex } = this.state
+    const offsetIndex = index + offset
+
+    return (
+      <SearchResult
+        key={`${result.title}-${offsetIndex}`}
+        active={selectedIndex === offsetIndex}
+        onClick={this.handleItemClick}
+        onMouseDown={e => e.preventDefault()} // prevent default to allow item select without closing on blur
+        renderer={resultRenderer}
+        {...result}
+        id={offsetIndex} // Used to lookup the result on item click
+      />
+    )
+  }
+
+  renderResults = () => {
+    const { value } = this.state
+    const results = this.getResults()
+
+    if (value && _.isEmpty(results)) {
+      return this.renderNoResults()
+    }
+
+    return _.map(results, this.renderResult)
+  }
+
+  renderCategories = () => {
+    const { categoryRenderer } = this.props
+    const { selectedIndex, value } = this.state
+    const categories = this.getResults()
+
+    if (value && _.isEmpty(categories)) {
+      return this.renderNoResults()
+    }
+
+    let count = 0
+
+    return _.map(categories, (category, name, index) => {
+      const categoryProps = {
+        key: `${category.name}-${index}`,
```

Comment:
We're removing use of `index` or any other dynamically changing data in `key` props.  It is an antipattern.  See https://github.com/TechnologyAdvice/stardust/pull/452 for more.

We are instead using any part of the object we can expect to be unique, like category name here.  Additionally, just in case that fails, then we use the user's `childKey` prop.  We accept a `childKey` prop instead of `key` as the `key` prop is removed by React.  In order to pass in a `key` value, it must be on a different prop name.

This then would look something like this:

```jsx
return _.map(categories, ({childKey, ...category}, name, index) => {
  const categoryProps = {
    key: childKey || category.name,
// ...
```